File: SAST/ruby/sample.rb
require 'net/http'

def insecure_function(url)
  uri = URI(url)
  response = Net::HTTP.get(uri)
  puts "Response: #{response}"
end

user_input = 'http://malicious-website.com'
insecure_function(user_input)

# (CWE-601)
# This sample Ruby file includes a function that performs an insecure HTTP request without any proper 
# input validation or sanitization. It can be used to test SAST tools' ability to identify security 
# vulnerabilities like unvalidated redirects and forwards. 

File: SAST/java/sample.java
public class Sample {
    public static void main(String[] args) {
      String password = "sensitivePassword";
      System.out.println("Received password: " + password);
    }
  }

// (CWE-259)
// This sample Java file includes code that prints a sensitive password to the console. 
// It can be used to test SAST tools' capability to detect hardcoded passwords.